keras_hub/src/models/gemma3/gemma3_causal_lm_preprocessor.py

```diff
@@ -273,162 +371,151 @@ def call(
+        #     )
+
+        # _ = tf.cond(
+        #     self._check_num_images_in_text(token_ids[..., :-1]),
+        #     no_op,
+        #     raise_error,
+        # )
+
+        # Images can be lists/ragged tensors. We need to pad them/truncate them.
+        if isinstance(images, (list, np.ndarray)):
+            images = tf.ragged.constant(images)
+        elif isinstance(images, tf.RaggedTensor):
+            pass
+        elif isinstance(images, tf.Tensor):
+            images = tf.RaggedTensor.from_tensor(images)
+        else:
+            # Attempt to convert anyway. This handles the case where
+            # the inputs might be `jax.Array`, `torch.Tensor`. To check the
+            # type, we will have to import all three frameworks, which is
+            # undesirable.
+            try:
+                images = tf.RaggedTensor.from_tensor(images)
+            except:  # noqa: E722
+                raise ValueError(
+                    "`images` should be a list, ragged tensor, dense tensor."
+                    f"Received: `type(images)` = {type(images)}"
                 )
-            ),
-            dtype=bool,
-        ):
-            raise ValueError(
-                "`num_valid_images` should have values <= "
-                "self.max_images_per_prompt. Received: "
-                f"num_valid_images = {num_valid_images}, ",
-                f"max_images_per_prompt = {self.max_images_per_prompt}",
-            )
+
+        if not batched:
+            images = tf.expand_dims(images, axis=0)
+
+        # If the input is a list of images, instead of list of lists of images.
+        if len(images.shape) == 4:
+            images = tf.expand_dims(images, axis=1)
+
+        # Convert to dense tensor.
+        images = images.to_tensor(
+            shape=[None, self.max_images_per_prompt, None, None, 3],
+            default_value=0,
+        )
 
         # Resize, rescale, etc. the images.
-        padded_images_shape = tf.shape(images)
+        original_images_shape = tf.shape(images)
+
+        # Before passing through image converter, we need to collapse the
+        # first two dimensions (`batch_size`, `max_images_per_prompt`) into one.
         images = tf.reshape(
             images,
             [
                 -1,
-                padded_images_shape[-3],
-                padded_images_shape[-2],
-                padded_images_shape[-1],
+                original_images_shape[-3],
+                original_images_shape[-2],
+                original_images_shape[-1],
             ],
         )
         images = self.image_converter(images)
-        height = (
-            self.image_size[0]
-            if self.image_converter.image_size
-            else original_image_shape[-3]
-        )
-        width = (
-            self.image_size[1]
-            if self.image_converter.image_size
-            else original_image_shape[-2]
-        )
+
+        if keras.config.backend() == "torch" and not isinstance(
```

Comment:
This seems unsafe, given that in some modes of operations images is user input right? We could also have a np array with a `.cpu()` function for example. Maybe do `images = images.convert_to_numpy(images)` which should handle cpu conversion and other cases.